site/src/components/WorkspaceQuota/WorkspaceQuota.tsx

```diff
@@ -0,0 +1,114 @@
+import Box from "@material-ui/core/Box"
+import LinearProgress from "@material-ui/core/LinearProgress"
+import { makeStyles } from "@material-ui/core/styles"
+import Skeleton from "@material-ui/lab/Skeleton"
+import { ErrorSummary } from "components/ErrorSummary/ErrorSummary"
+import { Stack } from "components/Stack/Stack"
+import { FC } from "react"
+import * as TypesGen from "../../api/typesGenerated"
+import { MONOSPACE_FONT_FAMILY } from "../../theme/constants"
+
+export const Language = {
+  of: "of",
+  workspace: "workspace",
+  workspaces: "workspaces",
+}
+
+export interface WorkspaceQuotaProps {
+  quota?: TypesGen.WorkspaceQuota
+  error: Error | unknown
+}
+
+export const WorkspaceQuota: FC<WorkspaceQuotaProps> = ({ quota, error }) => {
+  const styles = useStyles()
+
+  // error state
+  if (error !== undefined) {
+    return (
+      <Box>
+        <Stack spacing={1} className={styles.stack}>
+          <span className={styles.title}>Workspace Quota</span>
+          <ErrorSummary error={error} />
+        </Stack>
+      </Box>
+    )
+  }
+
+  // loading
+  if (quota === undefined) {
+    return (
+      <Box>
+        <Stack spacing={1} className={styles.stack}>
+          <span className={styles.title}>Workspace Quota</span>
+          <LinearProgress color="primary" />
+          <div className={styles.label}>
+            <Skeleton className={styles.skeleton} />
+          </div>
+        </Stack>
+      </Box>
+    )
+  }
+
+  // don't show if limit is 0, this means the feature is disabled.
+  if (quota.user_workspace_limit === 0) {
+    return <></>
```

Comment:
If there is nothing to render, we should be able to return `null`:

```suggestion
    return null
```